### DISCUSSION TITLE
POSIBILITY

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#741] [Q&A] [answered]
Is this possible  for encoders or SBC encoder to get compress bytes , 

For example  I don't want the  encoded data to write to file but  need it in array  for further processing  ??

encoder.write(samples, bytes_wrte);---->  this  write to file , i want to write It to array not file ??

**Comment by pschatzmann** [comment, marked as answer]
If I remember right you already asked this question and the answer is still the same:
use a CallbackStream or a RingBufferStream as output.
In the CallbackStream you can register a callback and in the RingBufferStream you can read the data that was converted

**Reply by aqildad-create** [reply]
thanks for the answer I will try theses streams to get converted data ,

and  Its first time I asked this question  and apology If  I mis   documentation .

**Reply by pschatzmann** [reply]
https://github.com/pschatzmann/arduino-audio-tools/discussions/725
